### PR TITLE
Fix `BookingRule` appearing twice

### PR DESCRIPTION
**Pull request by VillePihlava**
**Summary:**

Fix `BookingRule` appearing twice and add test for duplicates

## Summary by CodeRabbit

* **Tests**
  * Added a unit test that validates the GTFS entity schema contains no duplicate entity entries, improving confidence in data integrity and preventing subtle runtime issues.

### CODE REVIEW

File: onebusaway-gtfs/src/test/java/org/onebusaway/gtfs/serialization/GtfsEntitySchemaFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  @Test
+  public void testGetEntityClassesHasNoDuplicates() {
+    List<Class<?>> classes = GtfsEntitySchemaFactory.getEntityClasses();
+    Set<Class<?>> seen = new HashSet<>();
+    List<Class<?>> duplicates =
+        classes.stream().filter(c -> !seen.add(c)).collect(Collectors.toList());
+    assertTrue(
+        duplicates.isEmpty(),
+        "getEntityClasses() must not contain duplicate entries: " + duplicates);
+  }
```
Comment (leonardehrenfried) [inline onebusaway-gtfs/src/test/java/org/onebusaway/gtfs/serialization/GtfsEntitySchemaFactoryTest.java]:
```suggestion
    assertThat(
        duplicates, "getEntityClasses() must not contain duplicates").isEmpty();
  }
```

Comment (leonardehrenfried) [reply onebusaway-gtfs/src/test/java/org/onebusaway/gtfs/serialization/GtfsEntitySchemaFactoryTest.java]: You need to clean this up a bit, but you get the idea.

Comment (VillePihlava) [reply onebusaway-gtfs/src/test/java/org/onebusaway/gtfs/serialization/GtfsEntitySchemaFactoryTest.java]: Its now, is it what you wanted?
```
assertThat(duplicates).isEmpty();
```

Comment (leonardehrenfried) [reply onebusaway-gtfs/src/test/java/org/onebusaway/gtfs/serialization/GtfsEntitySchemaFactoryTest.java]: Is fine.

**Review by leonardehrenfried — APPROVED** [review]
